docs/content/concepts/assets/asset-observations.mdx

```diff
@@ -0,0 +1,105 @@
+
+## Relevant APIs
+
+| Name                                   | Description                                                          |
+| -------------------------------------- | -------------------------------------------------------------------- |
+| <PyObject object="AssetObservation" /> | Dagster event indicating that an asset's metadata has been recorded. |
+| <PyObject object="AssetKey" />         | A unique identifier for a particular external asset.                 |
+
+## Overview
+
+<PyObject object="AssetObservation" /> events are used record metadata in Dagster
+about a given asset. Asset observation events can be yielded at runtime within ops
+and assets. An asset must be defined using the <PyObject
+object="asset"
+decorator
+/> decorator or have existing materializations in order for its observations to be
+displayed.
+
+## Yielding an AssetObservation from an Asset
+
+To make Dagster aware that we have recorded metadata about an asset, we can yield an <PyObject object="AssetObservation" /> event:
+
+```python file=/concepts/assets/observations.py startafter=start_observation_asset_marker_0 endbefore=end_observation_asset_marker_0
+from dagster import AssetObservation
+
+
+@op
+def observation_op():
+    df = read_df()
+    remote_storage_path = persist_to_storage(df)
+    yield AssetObservation(asset_key="observation_asset", metadata={"num_rows": len(df)})
+    yield AssetMaterialization(
+        asset_key="observation_asset", description="Persisted result to storage"
+    )
+    yield Output(5)
+```
+
+We should now see an observation event in the event log when we execute this asset.
+
+<Image
+alt="asset-observation"
+src="/images/concepts/assets/observation.png"
+width={1417}
+height={917}
+/>
+
+### Attaching Metadata to an AssetObservation
+
+There are a variety of types of metadata that can be associated with an observation event, all through the <PyObject object="EventMetadataEntry" /> class. Each observation event optionally takes a list of metadata entries that are then displayed in the event log and the [Asset Details](/concepts/dagit/dagit#asset-details) page. Check our API docs for <PyObject object="EventMetadataEntry" /> for more details on the types of event metadata available.
+
+```python file=concepts/assets/observations.py startafter=start_observation_asset_marker_2 endbefore=end_observation_asset_marker_2
+from dagster import op, AssetObservation, Output, EventMetadata
+
+
+@op
+def observes_dataset_op():
+    df = read_df()
+    remote_storage_path = persist_to_storage(df)
+    yield AssetObservation(
+        asset_key="my_dataset",
+        metadata={
+            "text_metadata": "Text-based metadata for this event",
+            "path": EventMetadata.path(remote_storage_path),
+            "dashboard_url": EventMetadata.url("http://mycoolsite.com/url_for_my_data"),
+            "size (bytes)": calculate_bytes(df),
+        },
+    )
+    yield AssetMaterialization(asset_key="my_dataset")
```

Comment:
Same comment as above: probably better to leave out the AssetMaterialization

Reply:
I can leave out the AssetMaterialization for the examples that don't link to the asset details page. This example links to asset details page and without the asset materialization, users may not be able to nav to the asset details page so I think we should keep it in